lib/utils/check-scoped-type.js

```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  isTypeError: function (objects, type) {
+    if (Array.isArray(objects)) {
+      return objects.some(function (object) { object.type && object.type !== type })
+    }
+
+    return typeof objects === 'object' && objects.type && objects.type !== type
+  },
+
+  createErrorMessage: function (type) {
+    return 'type field in document does not match scoped store type of \'' + type + '\''
+  }
+}
```

Comment:
I decided to centralize where the error message is being created.  I think that this strikes a nice balance between code readability and DRY.  This way, the individual methods only need to check if there's a problem and then call `createErrorMessage(type)` and, should we decide to change the criteria for a scoped type conflict or change the content of the error message, we still only need to do that in one place.